ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -806,15 +812,17 @@ do
     if [[ "${UPGRADE_CCO_MANUAL_MODE}" == "oidc" ]]; then
         update_cloud_credentials_oidc
     fi
+
     upgrade
     check_upgrade_status
```

Comment:
I think about this step again, in the first time, we may exit `check_upgrade_status` a bit earlier when `UPGRADE_RHEL_WORKER_BEFOREHAND` enabled, then rhel worker get started to upgrade, in the meantime, the rhcos cluster is also ongoing to upgrade, if rhel worker upgrade is completed prior to rhcos cluster upgrade (because machineconfig might take more time than rhel worker upgrade), once that, maybe we need to run `check_upgrade_status` again (need to reset `UPGRADE_RHEL_WORKER_BEFOREHAND` to false in the 2nd time) to wait the whole cluster totally complete the upgrade. WDYT?

Reply:
yes, that would be more reliable for waiting the cluster to be ready. I added another check 
 `if [[ "${UPGRADE_RHEL_WORKER_BEFOREHAND}" == "triggered"` after RHEL worker upgrade, and then run `check_upgrade_status` again

Reply:
@jianlinliu yesterday I had another check on the openshift-ansible code, it turns out it's using the `desired.version` to get the expected cri-o/openshift-kubelet package version, 
https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_node/tasks/install.yml#L47 
which means we don't even have to wait for cluster version to get updated.

In the rehearse job https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/49932/rehearse-49932-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-4.16-upgrade-from-stable-4.15-azure-ipi-workers-rhel8-f28/1772965007206649856

```
Running command: oc get clusterversion
NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.15.3    True        True          9m45s   Unable to apply 4.16.0-0.nightly-2024-03-25-100907: an unknown error has occurred: MultipleErrors
Upgrade stuck at updating RHEL worker, run the RHEL worker upgrade now...
```
It start the RHEL upgrade after `9m45s` because the keywords `Unable to apply 4.16.0-0.nightly-2024-03-25-100907` detected. The job failed for a repo connection flake, but anyway the 4.15 to 4.16 RHEL worker upgrade is still being blocked by https://issues.redhat.com/browse/OCPBUGS-30802, so we can't get a successful rehearse test currently.

Reply:
good to know.